moto/cloudtrail/models.py

```diff
@@ -318,7 +318,12 @@ def describe_trails(self, include_shadow_trails):
         if include_shadow_trails:
             current_account = cloudtrail_backends[self.account_id]
             for backend in current_account.values():
-                all_trails.extend(backend.trails.values())
+                for trail in backend.trails.values():
+                    if (
+                        trail.is_multi_region
+                        or trail.region_name == backend.region_name
```

Comment:
```suggestion
                        or trail.region_name == self.region_name
```
The current `trail` is coming from `backend`, so this will always return True at the moment.

Would you mind adding a test case for this as well?
We already have `tests/test_cloudwatch/test_cloudwatch.py::test_describe_trails_without_shadowtrails` that tests this functionality, but I'm expecting that to fail with this fix. Ideally we cover all all possibilities in the existing tests, i.e. we create trails with MultiRegion=True and with MultiRegion=False, in multiple regions, and then verify only the correct trails are returned.

Reply:
Hi @bblommers, you're right. I've modified the CloudTrail test to address this change, because with this new logic the MultiRegion/NotMultiRegion and Home trails are correctly handled now.

Thanks!

Reply:
We have not reviewed the whole cloudtrail service, but maybe this issue is affecting other parts.
It is really tricky how AWS handle the `describe-trails` api call when listing multiregion and organization trails.
If you need help let us know.

Thanks !!